libs/langchain/langchain/retrievers/parent_document_retriever.py

```diff
@@ -56,19 +57,47 @@ class ParentDocumentRetriever(MultiVectorRetriever):
             )
     """
 
-    child_splitter: TextSplitter
-    """The text splitter to use to create child documents."""
+    child_splitter: Optional[TextSplitter] = None
+    """The text splitter to use to create child documents.
+    If none, you will have to provide a list of text splitters corresponding
+    to each document when calling `add_documents`."""
 
+    id_key: str = "doc_id"
     """The key to use to track the parent id. This will be stored in the
     metadata of child documents."""
+
     parent_splitter: Optional[TextSplitter] = None
     """The text splitter to use to create parent documents.
     If none, then the parent documents will be the raw documents passed in."""
 
+    index_args: Optional[Dict[str, Any]] = None
+    """If defined, indexing will be used when storing split documents to the
+    vectorstore. This should contain the arguments that you would pass to
+    `index`, but without `docs_source` and `vector_store`, both of which will
+    be passed automatically."""
+
+    @property
```

Comment:
a property that returns a function is a bit weird - shouldnt this just be a method?

Reply:
You're right. I'm not sure what I was originally thinking here, but as it stands, this is just a method with extra steps. I'll fix it.